src/include/addr_parsing.h

```diff
@@ -14,6 +14,9 @@
 extern "C" {
 #endif
 
+/*
+ * returns a number which allows to know if cat is safe for the memory
```

Comment:
The reality of what the function does is different. Your comment makes it look like it is a predicate. Could you please use the doxygen convention ?